crates/bevy_render/src/view/mod.rs

```diff
@@ -90,26 +91,141 @@ pub struct ViewUniformOffset {
+
+    pub fn new(
+        render_device: &RenderDevice,
+        texture_cache: &mut TextureCache,
+        msaa: &Msaa,
+        size: UVec2,
+        out_texture: TextureView,
+        hdr: bool,
+    ) -> ViewTarget {
+        let size = Extent3d {
+            width: size.x,
+            height: size.y,
+            depth_or_array_layers: 1,
+        };
+
+        let main_texture_format = match hdr {
+            true => ViewTarget::TEXTURE_FORMAT_HDR,
+            false => TextureFormat::bevy_default(),
+        };
+
+        let main_texture = texture_cache.get(
+            render_device,
+            TextureDescriptor {
+                label: Some("main_texture"),
+                size,
+                mip_level_count: 1,
+                sample_count: 1,
+                dimension: TextureDimension::D2,
+                format: main_texture_format,
+                usage: TextureUsages::RENDER_ATTACHMENT | TextureUsages::TEXTURE_BINDING,
             },
-            resolve_target: if self.sampled_target.is_some() {
-                Some(&self.view)
-            } else {
-                None
+        );
+
+        let sampled_main_texture = (msaa.samples > 1).then(|| {
+            texture_cache
+                .get(
+                    render_device,
+                    TextureDescriptor {
+                        label: Some("main_texture_sampled"),
+                        size,
+                        mip_level_count: 1,
+                        sample_count: msaa.samples,
+                        dimension: TextureDimension::D2,
+                        format: main_texture_format,
+                        usage: TextureUsages::RENDER_ATTACHMENT,
+                    },
+                )
+                .default_view
+        });
+
+        let main_texture = if hdr {
+            let ldr_texture = texture_cache.get(
+                render_device,
+                TextureDescriptor {
+                    label: Some("ldr_texture"),
+                    size,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::bevy_default(),
+                    usage: TextureUsages::RENDER_ATTACHMENT | TextureUsages::TEXTURE_BINDING,
+                },
+            );
+
+            ViewMainTexture::Hdr {
+                hdr_texture: main_texture.default_view,
+                sampled_hdr_texture: sampled_main_texture,
+                ldr_texture: ldr_texture.default_view,
+            }
+        } else {
+            ViewMainTexture::NoHdr {
+                texture: main_texture.default_view,
+                sampled_texture: sampled_main_texture,
+            }
+        };
+
+        ViewTarget {
+            main_texture,
+            out_texture,
+        }
+    }
+
+    pub fn get_color_attachment_hdr(&self, ops: Operations<Color>) -> RenderPassColorAttachment {
```

Comment:
I feel like the renaming of this is unnecessary/gives an incorrect representation of what it does. The returned colour attachment could be either HDR or SDR. It makes the diff smaller to leave it named as it was too. :)